ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/config/Cli.scala

```diff
@@ -229,56 +234,33 @@ object Cli {
         config.fold(error => throw new IllegalArgumentException(error), identity)
       }
 
-    private def filters(listOfIds: String): Either[String, Map[String, Option[List[Identifier]]]] =
+    private def filters(
+        listOfIds: String
+    ): Either[String, List[WorkflowConfig.StreamConfig.PartyFilter]] =
       listOfIds
         .split('+')
         .toList
         .map(filter)
-        .foldLeft[Either[String, Map[String, List[Identifier]]]](Right(Map.empty)) {
-          case (acc, next) =>
-            for {
-              filters <- acc
-              filter <- next
-            } yield filters + filter
-        }
-        .map { filters =>
-          filters.map { case (party, templateIds) =>
-            party -> Some(templateIds).filter(_.nonEmpty)
-          }
+        .foldLeft[Either[String, List[WorkflowConfig.StreamConfig.PartyFilter]]](
+          Right(List.empty)
+        ) { case (acc, next) =>
+          for {
+            filters <- acc
+            filter <- next
+          } yield filters :+ filter
         }
 
-    private def filter(filterString: String): Either[String, (String, List[Identifier])] =
+    private def filter(
+        filterString: String
+    ): Either[String, WorkflowConfig.StreamConfig.PartyFilter] = {
       filterString
         .split('@')
         .toList match {
         case party :: templates =>
-          templates
-            .map(templateIdFromString)
-            .foldLeft[Either[String, List[Identifier]]](Right(List.empty[Identifier])) {
-              case (acc, next) =>
-                for {
-                  ids <- acc
-                  id <- next
-                } yield id :: ids
-            }
-            .map(party -> _)
+          Right(WorkflowConfig.StreamConfig.PartyFilter(party, templates))
```

Comment:
Template ids are now `String`s to allow flexible filter specification through the YAML file (one can define a template in the short form like: `Foo1`).